cmd/root.go

```diff
@@ -83,6 +84,28 @@ func Run(spec *Specification) (*event.EventEmitter, error) {
 	// Set Time from ntp
 	network.NtpDate()
 
+	hw, err := n.ReadHardwareDetails()
+	if err != nil {
+		return eventEmitter, errors.Wrap(err, "unable to read all hardware details")
+	}
+
+	m, err := hammer.fetchMachine(spec.MachineUUID)
+	if err != nil {
+		return eventEmitter, nil
+	}
+
+	if m != nil && m.Allocation != nil && m.Allocation.Reinstall != nil {
```

Comment:
shouldnt this be if m != nil && m.Allocation != nil && m.Allocation.Reinstall

Reply:
The Reinstall field is of type ModelsV1MachineReinstall and contains information needed for reinstall abortion. If reinstall is triggered, this field is != nil, otherwise == nil